### DISCUSSION TITLE
This cluster should not be updated to the next minor version.

### DISCUSSION THREAD

**Discussion by eliassteiner** [okd-project/okd#1820] [General]
During an attempt to update our OpenShift cluster from version 4.13.0-0.okd-2023-10-28-065448 to the latest version, 4.14.0-0.okd-2023-11-14-101924.

The update process triggered the following warning alert:
```
Warning alert: This cluster should not be updated to the next minor version.
Cluster operator kube-apiserver should not be upgraded between minor versions: KubeletMinorVersionUpgradeable: Kubelet minor versions on 6 nodes will not be supported in the next OpenShift minor version upgrade.
```

Is there any possibility to update the kube-apiserver?
kind regards

**Comment by vrutkovs** [comment]
If the upgrade has already started this message is expected. Your upgrade is still going and nothing wrong with it.

What is says is that kube-apiserver is being upgraded to 1.27 and your nodes have kubelet at 1.26. If you abort the upgrade now and start upgrade to 4.15 OKD (kube 1.28) this won't be supported, as you'll skip a version of kube-apiserver and end up with kubelet two versions behind expected (1.26 -> 1.28).

This is not supported and this alerts warns you against aborting upgrade now - it expects you to either wait for it complete or restore from etcd backup if things went wrong.

**Reply by eliassteiner** [reply]
Thank you for your response. The update is frozen. I focused on the error message during the search. In this case, it is now resolved after identifying the problem. Thanks a lot.

**Comment by eliassteiner** [comment]
It seems like you're facing two issues related to the kube-api server in your Kubernetes/OpenShift cluster. Let me translate the provided information into English:

Issue 1 (Master-02):

The kube-api server on master-02 is in the "Init:CrashLoopBackOff" state.
There are no visible events or logs available for the pod.
Issue 2 (Master-01):

The kube-api server on master-01 is not starting.
Output from the oc get clusteroperator -o yaml kube-apiserver command shows several conditions:
Degraded Condition:
Last Transition Time: "2023-11-30T09:04:47Z"
Message: 'GuardControllerDegraded: Missing operand on node master-01.okd.bla-lab.ch'
Reason: GuardController_SyncError
Status: "True"
Type: Degraded
Progressing Condition:
Last Transition Time: "2023-11-23T05:34:44Z"
Message: 'NodeInstallerProgressing: 2 nodes are at revision 0; 1 node is at revision 58; 0 nodes have achieved new revision 67'
Reason: NodeInstaller
Status: "True"
Type: Progressing
Available Condition:
Last Transition Time: "2023-06-26T07:55:23Z"
Message: 'StaticPodsAvailable: 1 node is active; 2 nodes are at revision 0; 1 node is at revision 58; 0 nodes have achieved new revision 67'
Reason: AsExpected
Status: "True"
Type: Available
Kubelet Minor Version Upgradeable Condition:
Last Transition Time: "2023-11-23T05:33:58Z"
Message: 'KubeletMinorVersionUpgradeable: Kubelet minor versions on 6 nodes will not be supported in the next OpenShift minor version upgrade.'
Reason: KubeletMinorVersion_KubeletMinorVersionUnsupportedNextUpgrade
It seems there is an issue with GuardController on master-01, and the NodeInstaller is not progressing as expected. Investigating the missing operand on master-01 in the GuardControllerDegraded condition could be a good starting point to resolve the problem.
